docs/ci-integration/ci-integration.md

```diff
@@ -0,0 +1,118 @@
+
+To install `docker`, use the most recent versions [directly from Docker](https://docs.docker.com/engine/install/#server). The versions packaged for many distributions tend to fall behind.
+
+#### macOS
+
+To install `git`, the easiest way is to install the "XCode Command Line Tools". If you open up `Terminal`, and type:
+
+```go
+git --version
+```
+
+Then macOS will prompt you to install these tools. You can also use the `git` provided installer or Homebrew, if you prefer. [Details can be found here](https://git-scm.com/download/mac).
+
+To install `docker`, [download and install Docker CE](https://hub.docker.com/editions/community/docker-ce-desktop-mac). Be sure to grab the correct installer depending on your CPU architecture.
+
+#### Windows
+
+To install `git`, use the  [MSI installer](https://gitforwindows.org/). This will provide `git`, and a Bash shell; which may prove more natural for using Earthly. You may also use your package manager of choice.
+
+To install `docker`, [download and install Docker CE](https://hub.docker.com/editions/community/docker-ce-desktop-windows). Both the HyperV and WSL2 backends are supported, but the WSL2 one is very likely to be faster.
+
+### Installation
+
+Once you have ensured that the dependencies are available, you'll need to install `earthly` itself.
+
+#### Bare Metal
+
+This is the simplest method for adding `earthly` to your CI. It will work best on dedicated computers, or in scripted/auto-provisioned build environments. You can follow our [regular installation guide](https://earthly.dev/get-earthly) to add `earthly` if desired.
+
+If you are going to mostly be working from a WSL2 prompt in Windows, you might want to consider following the Linux instructions for installation. This will help prevent any cross-subsystem file transfers and keep your builds fast. Note that the "original" WSL is unsupported.
+
+If your build host is persistent, it is recommended to install `earthly` as part of the new host's configuration, and not as part of your build. This will speed up your builds, since you do not need to download `earthly` each time; and it will also provide stability in case a future version of `earthly` changes the behavior of a command.
+
+#### Containers
+
+Earthly currently offers two official images:
+
+- [`earthly/earthly`](https://hub.docker.com/r/earthly/earthly), which is a 1-stop shop. It includes a built-in `earthly-buildkitd` daemon, and accepts a target to be built as a parameter. It requires a mount for your source code, and an accessible `DOCKER_HOST`. When building a runner image for your CI; it is usually easier to start from this image, and add the pieces you need. See [this Jenkins agent configuration](https://github.com/earthly/ci-examples/blob/ce20840cffd2a8b04a8bd5dce477751adac3f490/jenkins/Earthfile#L48-L54) for an example.
```

Comment:
> it is usually easier to start from this image, and add the pieces you need

Wondering if our suggestions should default to the assumption that nothing more is needed other than that image itself -- i.e. you don't need to extend the image, it already contains everything it needs. If you need more stuff, just download it via Earthly itself, not by expanding Earthly's image.

WDYT?

Reply:
Its hard to make that assumption; it only holds if the runner is invoking our container directly (Like GHA, Argo, other container-orchestrating CIs). If the CI wants to run a container as a runner, it falls apart (TeamCity, Jenkins... the older breeds).

Reply:
I hear you, but the alternative (right now) is also... less than ideal. They would need to either:

    Provide a live docker daemon (DinD or mounted from elsewhere) to allow earthly to run buildkitd, with privileged mode
    A remote buildkitd instance (experimental)

We could also solve this by moving buildkitd out of a container; but that would be Linux only, and create two ways of doing things. If we did allow that, perhaps an install script to set it up and configure it within someone elses container would be in order? But were going off on a tangent; we don't have that right now. Nor am I sure that its the right thing to do.

Admittedly, our image tag pickings are thin, and purposefully so right now. I didn't want to bless too many image tags & configurations without users needing them. Wanted to gather feedback and see what part was most important to push on next.

I think that building off of ours is the easiest approach given the current lay of the land. Hopefully, the remote buildkitd (and maybe? a buildkitd installer?) can eventually clean this up and adding earthly to a container is just as easy as a wget...

Reply:
Makes sense. Maybe we can share a link to the dockerfile, and they can crib off it if they can't use it themselves.

Reply:
I think I cleaned this up some? Let me know.